test/reg-test/fam-api-reg/fam_arithmetic_atomics_mt_reg_test.cpp

```diff
@@ -0,0 +1,772 @@
+/*
+ * fam_arithmetic_atomics_mt_reg_test.cpp
+ * Copyright (c) 2019 Hewlett Packard Enterprise Development, LP. All rights
+ * reserved. Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * See https://spdx.org/licenses/BSD-3-Clause
+ *
+ */
+/* Test Case Description: Tests fam_add/fam_sub and certain fam_atomic
+ * operations for multithreaded model.
+ * (test case has been commented as there are some issues with negative test
+ * scenarios with non-blocking calls.
+ */
+#include <fam/fam.h>
+#include <fam/fam_exception.h>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <limits.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "common/fam_test_config.h"
+
+using namespace std;
+using namespace openfam;
+
+fam *my_fam;
+Fam_Options fam_opts;
+
+Fam_Region_Descriptor *testRegionDesc;
+const char *testRegionStr;
+
+typedef struct {
+    Fam_Descriptor *item;
+    uint64_t offset;
+    int32_t tid;
+    int32_t msg_size;
+} ValueInfo;
+
+#define NUM_THREADS 10
```

Comment:
Shouldnt we reduce thread count ? The test time will increase in smaller machines. Can we make it 4 threads?

Reply:
As multi-threaded test cases have been simplified, time taken to run these tests have reduced.
So plan is to retain 10 as thread count.